src/storage/table/row_group.cpp

```diff
@@ -648,9 +676,125 @@ void RowGroup::MergeStatistics(idx_t column_idx, BaseStatistics &other) {
 	stats[column_idx]->statistics->Merge(other);
 }
 
-RowGroupPointer RowGroup::Checkpoint(TableDataWriter &writer, vector<unique_ptr<BaseStatistics>> &global_stats) {
+void RowGroup::ScanSegments(const std::function<void(Vector &, idx_t)> &callback, Vector &intermediate,
+                            idx_t column_idx) {
+	Vector scan_vector(intermediate.GetType(), nullptr);
+	for (auto segment = (ColumnSegment *)columns[column_idx]->data.root_node.get(); segment;
+	     segment = (ColumnSegment *)segment->next.get()) {
+		ColumnScanState scan_state;
+		scan_state.current = segment;
+		segment->InitializeScan(scan_state);
+
+		for (idx_t base_row_index = 0; base_row_index < segment->count; base_row_index += STANDARD_VECTOR_SIZE) {
+			scan_vector.Reference(intermediate);
+
+			idx_t count = MinValue<idx_t>(segment->count - base_row_index, STANDARD_VECTOR_SIZE);
+			scan_state.row_index = segment->start + base_row_index;
+
+			columns[column_idx]->CheckpointScan(segment, scan_state, start, count, scan_vector);
+
+			callback(scan_vector, count);
+		}
+	}
+}
+bool CompressionForTypeExists(PhysicalType type) {
+	if (type == PhysicalType::BOOL || type == PhysicalType::INT8 || type == PhysicalType::INT16 ||
+	    type == PhysicalType::INT32 || type == PhysicalType::INT64 || type == PhysicalType::INT128 ||
+	    type == PhysicalType::UINT8 || type == PhysicalType::UINT16 || type == PhysicalType::UINT32 ||
+	    type == PhysicalType::UINT64 || type == PhysicalType::FLOAT || type == PhysicalType::DOUBLE ||
+	    type == PhysicalType::LIST || type == PhysicalType::INTERVAL || type == PhysicalType::BIT ||
+	    type == PhysicalType::VARCHAR) {
+		return true;
+	}
+	return false;
+}
+
+CompressionType RowGroup::DetectBestCompressionMethod(idx_t &compression_idx, idx_t &col_idx,
+                                                      CompressionType compression_type) {
+	if (compression_type != CompressionType::COMPRESSION_AUTO) {
```

Comment:
This looks like a lot of code copied from the `ColumnDataCheckpointer`, can we clean this up and unify the code there?